torch/distributed/_tensor/api.py

```diff
@@ -222,6 +222,14 @@ def __new__(
                 "To construct DTensor from torch.Tensor, it's recommended to "
                 "use local_tensor.detach() and make requires_grad consistent."
             )
+        if isinstance(local_tensor, torch._subclasses.fake_tensor.FakeTensor) and any(
+            isinstance(x, torch.SymInt) for x in local_tensor.shape
```

Comment:
wondering can we put this logic into `__tensor_unflatten__` instead? `__new__` are used by runtime ops so if we can put the checks into `__tensor_unflatten__` that would be better